common/lib/dependabot/shared_helpers.rb

```diff
@@ -135,14 +134,18 @@ def self.escape_command(command)
         env: T.nilable(T::Hash[String, String]),
         stderr_to_stdout: T::Boolean,
         allow_unsafe_shell_command: T::Boolean,
-        error_class: T.class_of(HelperSubprocessFailed)
+        error_class: T.class_of(HelperSubprocessFailed),
+        command_type: Symbol,
+        timeout: Integer
       )
         .returns(T.nilable(T.any(String, T::Hash[String, T.untyped], T::Array[T::Hash[String, T.untyped]])))
     end
     def self.run_helper_subprocess(command:, function:, args:, env: nil,
                                    stderr_to_stdout: false,
                                    allow_unsafe_shell_command: false,
-                                   error_class: HelperSubprocessFailed)
+                                   error_class: HelperSubprocessFailed,
+                                   command_type: :network,
```

Comment:
We set `:network` that is 120 seconds as default timeout. 

Note that timeout is not for the full process of the command instead it is timeout for the command to either respond to std out or std errr.

Reply:
TBH this feels more complex/confusing than it needs to be.  This introduces magic symbols that callers must know to use, where those symbols are mapped to specific values in the new command helpers class.  

There's also some complexity coming from the addition of both command type and timeout arguments.  As a caller if I didn't know the internals of this function I would be confused that a command that was run with `timeout: -1` would have a timeout applied because `command_type: :network` has a special meaning.  Is timeout an explicit value?  An implicit value?  I need to dig into the code to know how to call this.

WDYT about creating constants for different common timeout values and having callers use those constants to set a timeout value when needed.  This removes having two params, makes the timeout field explicit, and maintains named common timeout values.


```ruby
module CommandTimeouts
   NETWORK = 120 # seconds
   LONG_RUNNING = 300 # seconds
   NONE = -1 # seconds
   LOCAL_TASK = 30 # seconds

   DEFAULT = NETWORK
end

def self.run_helper_subprocess(... timeout: CommandTimeouts::DEFAULT...)
```

Reply:
It is fixed. Please let me know if you see any issue now.